osquery/core/http.h

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sstream>
+
+#include <boost/property_tree/json_parser.hpp>
+
+namespace pt = boost::property_tree;
+
+namespace osquery{
+
+	struct curl_string {
+		char *ptr;
+		size_t len;
+	};
```

Comment:
let's try to not have any curl related artifacts in the headers. i know this isn't really a curl thing, but let's use idiomatic C++11 as much as possible.